HHS/OPRE-OPS
# adds checks to edit BLIs

## What changed
- allows BLIs that are  in DRAFT and IN_REVIEW and PLANNED to be edited from Agreement detail BLI table
- updates prop names to more generic


Does Not Include:
- Backend changes to reflect BLI status back to DRAFT
- e2e tests
- checks to see if you are authorized

## Issue

#1374 

## How to test

- goto Agreement 2 // has BLIs > IN_Review
- add a DRAFT BLI and send to review
- try to edit that BLI // it should allow you to
- save the changes and should see success Alert
- TODO: Backend changes coming to reflect BLI status back to DRAFT
